markovbot/commands.py

```diff
@@ -44,3 +47,71 @@
     await markovbot.say(sentence)
 
 
+@markovbot.command(pass_context=True, help='Mock the last specified user message.')
+async def mock(context, user: str = None):
+    server_context = context.server_context
+
+    if not server_context.is_ready:
+        await markovbot.say('I am still learning from all your messages. Try again later.')
+        return
+
+    if user is None:
+        log.debug('Bot issued say command on server %s',
+                  server_context.server.name)
+        sentence = server_context.markov.make_sentence_server()
+
+        if sentence is None:
+            await markovbot.say('Unable to generate message. This is probably due to a lack of messages on the server.')
+            return
+
+    else:
+        logs = markovbot.logs_from(server_context.message.channel, limit=20)
+        sentence = logs[-1]
```

Comment:
If you are trying to get a users last message then you will need to actually filter the logs based on the user id. This will just get the last message sent to the server. 

logs_from also returns an array of Message objects, not just strings.  It also is a coroutine, so it needs a yield in front of it.